rust/lance-linalg/src/kmeans.rs

```diff
@@ -538,12 +539,114 @@ impl KMeans {
     }
 }
 
+/// Return a slice of `data[x,y..y+strip]`.
+#[inline]
+fn get_slice(data: &[f32], x: usize, y: usize, dim: usize, strip: usize) -> &[f32] {
+    &data[x * dim + y..x * dim + y + strip]
+}
+
+/// Fast partition computation for L2 distance.
+fn compute_partitions_l2(centroids: &[f32], data: &[f32], dim: usize) -> Vec<u32> {
```

Comment:
nit: `compute_partitions_l2_f32` or `compute_partitions_l2<f32>`

Reply:
ok, lemme make it `_f32` first, get things going.